### ISSUE TITLE
[BUG] SKR Mini E3 V2.0 FAST_PWM_FAN ignores FAST_PWM_FAN_FREQUENCY (f=1kHz)

### ISSUE THREAD

**Issue by blazewicz** [MarlinFirmware/Marlin#22375]
### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Part cooling fan emits loud 1 kHz noise, investigation shown that PWM frequency is exactly 1 kHz. This doesn't change with `FAST_PWM_FAN_FREQUENCY`.

![IMG_20210716_215246](https://user-images.githubusercontent.com/13185945/126001820-2cfdcb4e-fe81-45e9-b050-45c7388c945d.jpg)

Board uses pin `PC6` for part cooling fan. This pin is channel 1 of timer `TIM3`. This board also has second PWM controlled fan output which is on pin `PC7` and its channel 2 of timer `TIM3`. (Arduino framework configures it to `TIM3`).

![Zrzut ekranu 2021-07-16 o 22 00 43](https://user-images.githubusercontent.com/13185945/126002661-903643d3-86fc-4ee8-a553-971c89430377.png)

~~Probably `TIM3` is used by some ISR which is set to 1 kHz before PWM is initialized.~~ Default value from `PWM_FREQUENCY` is used.

### Expected behavior

PWM frequency is configurable with `FAST_PWM_FAN_FREQUENCY`.

### Actual behavior

PWM frequency doesn't change and is always 1 kHz.

### Steps to Reproduce

Build with example config for Ender 3 with BTT SKR Mini E3 V2.0, set  `FAST_PWM_FAN` and `FAST_PWM_FAN_FREQUENCY` settings.

### Version of Marlin Firmware

bugfix-2.0.x

### Printer model

doesn't matter

### Electronics

BTT SKR Mini E3 V2.0

**Comment by blazewicz** [comment]
adding

```diff
--- a/buildroot/share/PlatformIO/variants/MARLIN_F103Rx/variant.h
+++ b/buildroot/share/PlatformIO/variants/MARLIN_F103Rx/variant.h
@@ -131,6 +131,8 @@ extern "C" {
 #define HAL_DAC_MODULE_ENABLED
 #endif

+#define PWM_FREQUENCY 31000
+
 #ifdef __cplusplus
 } // extern "C"
 #endif
```

does change fan PWM frequency

**Comment by thisiskeithb** [comment]
Please attach your configuration files.

Also, what is the exact MCU on the board and which PIO environment are you compiling with?

**Comment by blazewicz** [comment]
Its GD32F103RET6 (STM32F103RET6 clone). I use `STM32F103RE_btt_USB`, also tested `STM32F103RC_btt_USB`, same behavior.

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/6834371/Configuration.zip)

**Comment by thinkyhead** [comment]
Four different boards use `MARLIN_F103Rx` so we cannot make this change unless the other boards would not be harmed. If only certain boards need the define, it should be okay to add it to those boards' environments with `-DPWM_FREQUENCY=31000` on the build flags.

**Comment by tpruvot** [comment]
Double check there is no other use of TIMER3... i think its TONE timer which change the frequency

**Comment by blazewicz** [comment]
Maybe fixed in #23125, needs verification.

**Comment by thisiskeithb** [comment]
Closing since the GD-based SKR Mini cannot be added here.
